### ISSUE TITLE
Rule request: no-improper-passed-nulls

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#357]
*For teams with a large code base, turning on `--strictNullChecks` can take a while. Having rules that do part of the work would help incrementally enabling it.*

Don't pass null or undefined to a function that doesn't accept it.

Bad:

```typescript
class Person {
    private name;
    public setName(name: string) {
        this.name = name;
    }
}

const person = new Person();
person.setName(undefined); // This should be an error
```

**Comment by JoshuaKGoldberg** [comment]
This is grander in scale than what a linter can do well. Addressing this would need access to the TypeScript language service.

I've started work on this in [TypeStat](https://github.com/JoshuaKGoldberg/TypeStat) instead.
